core/src/main/java/com/linecorp/armeria/internal/client/DefaultDnsNameResolver.java

```diff
@@ -140,9 +148,8 @@ private void configureTimeout(List<DnsQuestion> questions, String logPrefix,
             final DnsTimeoutException exception = new DnsTimeoutException(
                     '[' + logPrefix + "] " + questions + " are timed out after " +
                     queryTimeoutMillis + " milliseconds.");
-            result.setFailure(exception);
             promises.forEach(promise -> {
-                promise.cancel(true);
```

Comment:
We should cancel the `promise` to stop sending DNS queries internally. https://github.com/netty/netty/pull/10171/files#diff-6107a6de600a6f941beb29f082976653R357
Is there any reason that `DnsTimeoutException` shouldn't be propagated to the caller?

Reply:
I change back to cancel, one reason I delete DnsTimeoutException in here is that it will trigger `operationComplete` with CancellationException. When we have A & AAAA queries, we may have A query sucess but AAAA query failed, it will call `aggregatedPromise.setSuccess` and throws IllegalStateException. So I move the `DnsTimeoutException` to `operationComplete`.

Reply:
Then how about just using `try(Success|failure)`? Becuase we can't be sure that the `CancellationException` is triggered by our side. (i.e. `DnsNameResolver` could cancel it internally)

Reply:
Originally I have the same idea..
But it can't allow one of the request success.
If one of A/AAAA success, we still return DnsTimeoutException. or we need to check all the future again here.

Reply:
Curious you mean tryFailure instead of canceling? or make listener using  `try(Success|failure)`?

Reply:
> or make listener using try(Success|failure)?

Yes, that's what I meant. 😄

Reply:
But then, we always failed when one of A/AAAA is succeed.

Reply:
Yes, as you said, if the query succeeded after the timeout, we are still going to get timeout, but I think it's ok. 😄

Reply:
I mean one of the query succeed, e.g. A is before timeout and AAAA is timeout.
Now I change to check if all of the query promises are done or not.

Reply:
> I mean one of the query succeeds before timeout, e.g. A is before timeout and AAAA is timeout.

Ah, I completely missed that. Thanks!